app/presenters/buyers/accounts_index_presenter.rb

```diff
@@ -1,24 +1,33 @@
 # frozen_string_literal: true
 
 class Buyers::AccountsIndexPresenter
-  def initialize(provider:, params: {})
+  include System::UrlHelpers.system_url_helpers
+
+  def initialize(provider:, user: nil, params: {})
     @provider = provider
     @pagination_params = { page: params[:page] || 1, per_page: params[:per_page] || 20 }
     @search = ThreeScale::Search.new(params[:search])
     @sorting_params = [params[:sort] || 'created_at', params[:direction] || 'desc']
+    @ability = Ability.new(user)
   end
 
-  attr_reader :provider, :pagination_params, :search, :sorting_params
+  attr_reader :provider, :pagination_params, :search, :sorting_params, :ability
 
   delegate :total_entries, to: :buyers
+  delegate :can?, to: :ability
 
   def buyers
-    @buyers ||= provider.buyer_accounts
-                        .includes([:admin_user])
-                        .not_master
-                        .scope_search(search)
-                        .order_by(*sorting_params)
-                        .paginate(pagination_params)
+    @buyers ||= raw_buyers.scope_search(search)
+                          .order_by(*sorting_params)
+                          .paginate(pagination_params)
+  end
+
+  def account_plans
+    @account_plans ||= provider.account_plans.stock
+  end
+
+  def account_plans_size
+    @account_plans_size ||= account_plans.length
```

Comment:
I think `.size` is more common for Ruby, but it's OK, they are doing the same thing for arrays.